app/components/search/SearchResults/SearchResults.tsx

```diff
@@ -77,8 +90,12 @@ const SearchResults = ({
                 <ClearSearchButton />
               </div>
             )}
-            {searchMapHitData.hits.map((hit: TransformedSearchHit) => (
-              <SearchResult hit={hit} key={`${hit.id} - ${hit.name}`} />
+            {searchMapHitData.hits.map((hit: TransformedSearchHit, index) => (
+              <SearchResult
+                hit={hit}
+                key={`${hit.id} - ${hit.name}`}
+                ref={index === 0 ? firstResultRef : null}
```

Comment:
I dove a little into React refs because I was curious how they work. It looks like in simple cases you can get a way with using a _callback ref_ -- a function that receives the current node as the argument -- instead of having to create a `firstResultRef` first. This would reduce some boilerplate. 

The type sig for `refs` show it can take either a `RefObject` or `RefCallback`. Neat!

```ts
type Ref<T> = RefCallback<T> | RefObject<T> | null;
```

Are you familiar with this technique? Check out: 
- https://tigeroakes.com/posts/react-focus-on-render/
- https://dev.to/gilfink/quick-tip-using-callback-refs-in-react-4gef
- https://react.dev/learn/manipulating-the-dom-with-refs#example-scrolling-to-an-element

I wonder if something like this would work automatically when the component tree is re-rendered on a page click. I am using `useCallback` because with an empty array dependency key it gives you the nicety of only calling itself on the first mount. But maybe worth experimenting without it too: 

```suggestion
                ref={index === 0 ? useCallback((node) => node?.focus(), []) : null}
```

Reply:
Nice !! Thanks for the suggestion